partiql-planner/src/test/kotlin/org/partiql/planner/internal/typer/functions/NullIfTest.kt

```diff
@@ -8,27 +8,34 @@ import org.partiql.planner.util.cartesianProduct
 import org.partiql.types.StaticType
 import java.util.stream.Stream
 
-// TODO: Model handling of Truth Value in typer better.
+/**
```

Comment:
iirc, SQL spec defines a rewrite for `COALESCE` using case-when. So we'd probably need to test `COALESCE`'s typing as well.

Reply:
Yes which we already rewrite.